tensorflow_addons/metrics/r_square.py

```diff
@@ -125,18 +132,25 @@ def update_state(self, y_true, y_pred, sample_weight=None) -> None:
             tf.reduce_sum((y_true - y_pred) ** 2 * sample_weight, axis=0)
         )
         self.count.assign_add(tf.reduce_sum(sample_weight, axis=0))
+        self.num_examples.assign_add(tf.size(y_true))
 
     def result(self) -> tf.Tensor:
         mean = self.sum / self.count
         total = self.squared_sum - self.sum * mean
         raw_scores = 1 - (self.res / total)
+        n = tf.cast(self.num_examples, dtype=tf.float32)
```

Comment:
It is better that you move the code in an internal function that converts R2 in an adjusted R2 so that it is computed only if it is required the adjusted version.